Спринт 15. Clean Architecture.

morechiba opened the pull request.

morechiba committed:
start Clean Architecture

morechiba committed:
Clean Architecture - продолжение, пока стоп

morechiba committed:
Clean Architecture - продолжение

morechiba committed:
правильные пути для activities
+поиск заработал частично (не работает плейсхолдер - noConnection)

morechiba committed:
поиск с обработкой кода ответа, работают все сценарии по ТЗ

morechiba committed:
убраны лишние зависимости
перенесены еще файлы по слоям

morechiba committed:
техническое, новая версия Kotlin

morechiba committed:
shared preferences частично
-не работает обновление списка "Вы искали" после возврата со страницы медиаплеера

morechiba committed:
shared preferences для переключения темы. Пока сломался поиск - не скрывается история с результатами поиска и не меняется порядок при выходе из плеера

morechiba committed:
переключение темы заработало правильно

morechiba committed:
заработал поиск правильно. Вынесла в отдельные состояния переключения видимостей - иначе все путается.

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/Creator.kt:
```diff
@@ -0,0 +1,50 @@
+
+object Creator {
+
+    lateinit var application: Application
```
Переменную лучше сделать приватной, так как уже есть отдельный сеттер, а наружу отдавать её не стоит

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/domain/impl/TrackHistoryInteractorImpl.kt:
```diff
@@ -1,18 +1,32 @@
+        val item = object : TypeToken<MutableList<Track>>() {}.type
+        val trackList: MutableList<Track> = gson.fromJson(searchHistory, item) ?: mutableListOf<Track>()
+        trackList.removeIf { it == track }
+        if (trackList.size == 10) {
```
Размер истории поиска лучше в константу вынести для повышения читабельности кода

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/presentation/App.kt:
```diff
@@ -0,0 +1,31 @@
+
+class App: Application() {
+
+    var darkTheme = false
```
Данной переменной лучше добавить `private set`, чтобы запретить изменение её значения извне, т.к. для изменения её значения написана отдельная функция

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/ui/SettingsActivity.kt:
```diff
@@ -29,7 +31,14 @@ class SettingsActivity : AppCompatActivity() {
         val themeSwitcher = findViewById<SwitchMaterial>(R.id.themeSwitcher)
         val app = applicationContext as App
-        themeSwitcher.isChecked = app.darkTheme
+        val theme = app.getSavedTheme()
```
Здесь нужно интерактор использовать, чтобы получать значение темы

ArturNurtdinov reviewed (commented):
Плейсхолдер обложки трека не соответствует макетам в списке треков:

<img width="854" height="152" alt="image" src="https://github.com/user-attachments/assets/bb41740d-95aa-46b0-8f64-241d9a477c7b" />

Макет:

<img width="1222" height="216" alt="image" src="https://github.com/user-attachments/assets/40ef2abb-5720-45de-94a5-31da16045576" />

morechiba committed:
1. правильный плейсхолдер для обложки трека в спике
2. переменная в Creator сделана приватной, так как уже есть отдельный сеттер, а наружу отдавать её не стоит
3. размер истории поиска вынесен в константу для повышения читабельности кода

morechiba committed:
Для переменной darkTheme добавлен private set, чтобы запретить изменение её значения извне (для изменения её значения написана отдельная функция)

morechiba committed:
1. убраны лишние зависимости
2. placeholder удален, чтоб не мешал, и окончательно исправлен для списка результатов поиска

morechiba committed:
получение значения темы через интерактор
